src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/HttpVersionTransformationAttribute.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+using System.Management.Automation;
+
+namespace Microsoft.PowerShell.Commands
+{
+    internal sealed class HttpVersionTransformationAttribute : ArgumentTransformationAttribute
+    {
+        public override object Transform(EngineIntrinsics engineIntrinsics, object inputData)
+        {
+            object version = PSObject.Base(inputData);
+
+            if (version is string versionStr && versionStr.Contains('.'))
+            {
+                // If the string contains a '.', let the Version constructor handle the conversion.
+                return inputData;
+            }
+
+            if (version is double versionDouble)
+            {
+                // 1.0 converts to string "1" so we preserve .0 with "0.0" format
+                return versionDouble.ToString("0.0", CultureInfo.InvariantCulture);
+            }
```

Comment:
Does this work for "-Version 2"? I guess we get the argument as int type.

Reply:
Do you mean `-HttpVersion 2`?
2 as int (handled bellow with other cases)
2.0 as double

Reply:
What if we implement a type converter so that LanguagePrimitives.ConvertTo() works for all needed cases? It seems then we can remove the transformation attribute and gets more benefits for users. 
Another thoughts we could implement the type converter and change the transformation attribute to process only well-known values like `Http10`, `Http20`, `Http11`, `Unknown `. It is for discussion.

I'd prefer the type converter and removing the attribute.

Reply:
Since we already have ArgumentToVersionTransformationAttribute I suggest to re-use it.

Thinking more about type converter I believe It can have undesirable side effects.